application/controllers/OptinController.php

```diff
@@ -54,13 +54,11 @@ function actiontokens($surveyid, $token, $langcode = '')
 
         Yii::app()->setLanguage($sBaseLanguage);
 
-        $aSurveyInfo = getSurveyInfo($iSurveyID, $sBaseLanguage);
-
-        if ($aSurveyInfo == false || !tableExists("{{tokens_{$iSurveyID}}}")) {
+        if (empty($oSurvey) || !$oSurvey->hasTokensTable) {
```

Comment:
empty test is not needed , done at https://github.com/TonisOrmisson/LimeSurvey/blob/b8af9b7256deb22e8ed8d8b3b7f67b86652182cc/application/controllers/OptinController.php#L42